// api-tools/taoanhroblox.js
// api-tools/taoanhroblox.js
// Giả lập tool tạo ảnh Roblox

module.exports = async (req, res) => {
    // Lấy params (dữ liệu) mà bot messenger gửi lên nè
    const { username, background } = req.query; // Thường là query params cho GET
    // Hoặc const { username, background } = req.body; // Thường là body cho POST

    // *** LOGIC TẠO ẢNH ROBOT VCL CỦA NÍ ĐẶT Ở ĐÂY NÈ ***
    
    if (!username) {
        return res.status(400).json({
            error: "Thiếu vcl cái username Roblox rùi ní ơi :((",
            cach_dung: "?username=tencuani&background=red",
            status: 400
        });
    }

    // Giả lập độ trễ tạo ảnh
    await new Promise(resolve => setTimeout(resolve, 500)); 

    // Gửi kết quả về cho bot messenger
    res.status(200).json({
        success: true,
        tool: "Tao Anh Roblox vcl",
        input: { username, background },
        image_url: `https://fakeapi.com/roblox_image_${username}_${Date.now()}.png`, // Link ảnh giả lập
        message: `Ảnh Roblox của **${username}** đã được tạo xong :))`
    });
};
